File: src/main/java/cn/spark/study/core/sql/SparkSessionTest.java
package cn.spark.study.core.sql;

import org.apache.spark.sql.AnalysisException;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import static org.apache.spark.sql.functions.col;
/**
 * Author:rzx
 * Date:2017/6/29
 */
public class SparkSessionTest {
    public static void main(String[] args) throws AnalysisException {
       SparkSession ss = SparkSession
               .builder()
               .appName("ss")
               .master("local")
               .config("asd","133")
               .getOrCreate();
       Dataset<Row> df = ss.read().json("input/people.json");
       df.printSchema();
       df.show();
       df.select(col("name"), col("age").plus(1)).show();
       df.filter(col("age").gt(20)).show();
       df.groupBy("age").count().show();

       // 注册一个DataFrame用sql临时视图
        df.createOrReplaceTempView("people");
        Dataset<Row> sqlRs =  ss.sql("select * from people");
        sqlRs.show();

        //创建一个全局的临时视图
        df.createGlobalTempView("people");
        //临时施图用于global_tem作为一个数据库保存
        Dataset<Row> sqlRsdb =ss.sql("SELECT * FROM global_temp.people");
        sqlRsdb.show();

    }
}

File: src/main/java/cn/spark/scala/core/sql/SparkSessionTest.scala
package cn.spark.scala.core.sql

import org.apache.spark.sql.SparkSession
/**
  * Author:rzx
  * Date:2017/6/29
  */

object SparkSessionTest {
  def main(args: Array[String]): Unit = {
    val sparkSession = SparkSession.builder.
      master("local")
      .appName("spark session example")
      .config("12312","123213")
      .getOrCreate()

    //导入隐式转换。。包。。。
    import sparkSession.implicits._

    val df =sparkSession.read.json("input/people.json")
    df.show()
    df.select("name","age").show(2)
    df.select(df("name"),df("age")+1).show(2)
    df.select($"name", $"age" + 1).show()
    df.filter($"age">20).show()
    //读取csv
    val dfCvs = sparkSession.read.csv("input/Advertising.csv")
    dfCvs.show()
    //读取csv设置头部
    val dfOptions = sparkSession.read.option("header","true").csv("input/Advertising.csv")
    dfOptions.show()

  }
}
